src/components/OptionToggle.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from "react";
+import Toggle from 'react-toggle';
+import "react-toggle/style.css";
+
+export interface OptionToggleProps {
+    optionLabel: string,
+    toggleFor: string
+}
+interface OptionToggleState {
+    setting: boolean,
+}
+interface ChromeSettingObject {
+    [key: string]: any
+}
+
+class OptionToggle extends React.Component<OptionToggleProps, OptionToggleState> {
```

Comment:
If the parameter `OptionToggleState` is not used, why is it separated from `OptionToggleProps`?

Reply:
State is used? It's used to store the value of the storage value. No?

Reply:
I never said `state` is not used; I said `OptionToggleState` is not used.

Reply:
According to typescript documentations I referred to without `OptionToggleState` state would not work. `Property 'setting' does not exist on type 'Readonly<{}>'.ts(2339)`

Reply:
OK, so `interface Component` has two optional parameters `P`(rops) and `S`(tates), and we need an object `S`(with `state`s we need as its keys) as the second parameter in order to use states.
https://stackoverflow.com/a/47562985/4370146

Reply:
Yeah this got me first time because in plain React.js I didn't had to provide parameters to Component to use states. Not that I am 100% sure my React coding style was up to the standards though.